charts/flyte-binary/gke-starter.yaml

```diff
@@ -0,0 +1,119 @@
+  storage:
+    # metadataContainer Bucket to store Flyte metadata
+    metadataContainer: "my-organization-flyte-container"
+    # userDataContainer Bucket to store Flyte user data
+    userDataContainer: "my-organization-flyte-container"
+    # provider Object store provider (Supported values: s3, gcs)
+    provider: gcs
+    # providerConfig Additional object store provider-specific configuration
+    providerConfig:
+      # gcs Provider configuration for GCS object store
+      gcs:
+        # project Google Cloud project in which bucket resides
+        project: "my-organization-gcp-project"
+  # logging Specify configuration for logs emitted by Flyte
+  logging:
+    # level Set the log level
+    level: 5
+    # plugins Specify additional logging plugins
+    plugins:
+      # stackdriver Configure logging plugin to have logs visible in StackDriver
+      stackdriver:
+        enabled: true
+        templateUri: |
+          https://console.cloud.google.com/logs/query;query=resource.labels.pod_name={{.podName}}&timestamp>{{.podRFC3339StartTime}}
+  # auth Specify configuration for Flyte authentication
+  auth:
+    # enabled Enable Flyte authentication
+    enabled: false
+    # oidc OIDC configuration for Flyte authentication
+    oidc:
+      # baseUrl URL for OIDC provider
+      baseUrl: ""
+      # clientId Flyte application client ID
+      clientId: ""
+      # clientSecret Flyte application client secret
+      clientSecret: ""
+    # internal Configuration for internal authentication
+    # The settings for internal still need to be defined if you wish to use an external auth server
+    # These credentials are used during communication between the FlyteAdmin and Propeller microservices
+    internal:
+      # clientId Client ID for internal authentication - set to flytepropeller or external auth server
+      clientId: flytepropeller
+      # clientSecret Client secret for internal authentication
+      clientSecret: ""
+      # clientSecretHash Bcrypt hash of clientSecret
+      clientSecretHash: ""
+    # authorizedUris Set of URIs that clients are allowed to visit the service on
+    authorizedUris: []
+
+  # inline Specify additional configuration or overrides for Flyte, to be merged with the base configuration
+  inline: 
+    #This section automates the IAM Role annotation for the default KSA on each project namespace to enable IRSA
+    #Learn more: https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html
+    cluster_resources:
+      customData:
+      - production:
+        - defaultIamServiceAccount:
+            value: <FLYTE_IAM_SA_EMAIL>
+      - staging:
+        - defaultIamServiceAccount:
+            value: <FLYTE_IAM_SA_EMAIL>
+      - development:
+        - defaultIamServiceAccount:
+            value: <FLYTE_IAM_SA_EMAIL>
+    plugins:
+      k8s:
+        inject-finalizer: true
+        default-env-vars:
+          - AWS_METADATA_SERVICE_TIMEOUT: 5
```

Comment:
I don't think these env vars are needed on GKE. Can we try without them?

Reply:
Also the `inject-finalizer`? Or just the `default-env vars` with AWS related things?

Reply:
@davidmirror-ops 
Also I noticed that there are no stackdriver logs, so I think the stackdriverTemplateUri needs to be updated. Any suggestions?

Reply:
What about the following structure:

```
templateUris:   
  - "https://console.cloud.google.com/logs/query;query=resource.labels.namespace_name%3D%22{{`{{.namespace}}`}}%22%0Aresource.labels.pod_name%3D%22{{`{{.podName}}`}}%22%0Aresource.labels.container_name%3D%22{{`{{.containerName}}`}}%22?project={{.GcpProjectId}}&angularJsUrl=%2Flogs%2Fviewer%3Fproject%3D{{.GcpProjectId}}"
```
Unf I don't have an environment to test it right now (it would take me some time to spin it up) but that config would typically be used with fluentbit.

Reply:
Also, please leave the `inject-finalizer` as it's needed for Garbage Collection purposes.

Reply:
Gonna give the suggested templateUri a try

Reply:
@davidmirror-ops 
The templateUri almost works. Only thing is that the `.GcpProjectId` isn't being populated in the template. Once fixed, the logging query does work (namespace, podname and container name do get populated correctly)

Reply:
Looked again at the AWS templateUri and realized that not everything is automatically filled in by Flyte, so made the Gcp Project Id in the templateUri something that the user has to fill in.